Fix/frontend fixes

u22520016 opened the pull request:
Hey guys...this is a big PR.  I am going to try to remember what I fixed.

1. I fixed all App bars to look the same and all "sub" pages' app bars are now smaller and have a color.  It fits some standards for app bars.
2. I redid the whole home page.  
3. You can now view the list of event invites and connection requests on a separate page.  There are tabs there to see connection requests and event invites
4. I added a way to directly go to the search page from the home page to make the home page more interesting
5. I changed the way it lets a user know that there are no event invites, friends online or connection requests.  It now shows a little graphic.
6. I changed the event page a bit.  I removed the top space when no events were upcoming or joined.  I also changed the tab buttons a bit.  I also removed the empty space at the bottom
7. I removed the reviews and genres from the game library page. I also added "Metacritic" to the Metacritic score and changed the position.
8. I added padding to the search bar and also added a clear button to clear search input text
9. I removed the space where the bio should be if there is no bio in the profile page.  I also changed the colors of the 3 buttons in the profile page.

I know its a lot of changes.  Please let me know if you find any errors or if something looks bad.  

PS:  I haven't work on the connection notifications yet.  I need to talk to you guys about that.

MonikaTheiss commented:
Hi @u22520016 , thank you so much for all of this! 
I am currently running your code on my side here are a few issues I found:

1. The appbars are confusing me, the appbar on the message page, updates and invites, and the search page when navigating to it through the homepage are all different.
2. Would it perhaps be possible to also , when the clear button on the search component is pressed, that the search stuff is cleared or is that too difficult?
3. Little worrisome issue, I cannot update my bio field any more... which seems to be an issue on other branches though
4.  For the updates and invites page,when you click on it , it first says no updates and then it updates and shows everything.. would it perhaps be possible to rather have a loading screen there?
5.  It might just be me, but the cards on the home page feel a bit weird to me. Would it perhaps look better if we made the background surface color and then rather add a primary border around the cards? Something on the home page does not sit right with me.



I really like the changes you made to the view events page and also the changes you made to the game library page👍 let me know what you think of the suggested changes.

u22520016 commented:
> Hi @u22520016 , thank you so much for all of this! I am currently running your code on my side here are a few issues I found:
> 
> 1. The appbars are confusing me, the appbar on the message page, updates and invites, and the search page when navigating to it through the homepage are all different.
> 2. Would it perhaps be possible to also , when the clear button on the search component is pressed, that the search stuff is cleared or is that too difficult?
> 3. Little worrisome issue, I cannot update my bio field any more... which seems to be an issue on other branches though
> 4. For the updates and invites page,when you click on it , it first says no updates and then it updates and shows everything.. would it perhaps be possible to rather have a loading screen there?
> 5. It might just be me, but the cards on the home page feel a bit weird to me. Would it perhaps look better if we made the background surface color and then rather add a primary border around the cards? Something on the home page does not sit right with me.
> 
> I really like the changes you made to the view events page and also the changes you made to the game library page👍 let me know what you think of the suggested changes.

1. Maybe I missed those, but some app bars are supposed to be different in the cases where a page is opened inside another page (so when a sub-page is opened, that app bar is not the suaul large and white text.  It is to indicate that the page is a sub-page)
2. It would, and I thought about it, but in my experience, sometimes when you make a spelling mistake, you would like to clear the search bar alone without clearing the results, so I did not think it would be good to clear all results everytime you click that button.  I could perhaps add a "clear search" button.  I'll have to look into some designs if we need to implement that.
3. I didn't not really change the bio functionality, so I'm not too sure about that.
4. I can add that yes.
5. I definitely see what you mean, I have have the same feeling.  I can maybe try to do that, but I changed it to at least give the home page some body.  I really do not know what else to do about that.  I could perhaps quickly ask for an expert opinion but I don't want to spend too much time on that.

MonikaTheiss commented on lib/view/pages/events/view_events_page.dart:
```diff
@@ -98,366 +96,163 @@ class _HomePageWidgetState extends State<ViewEvents> {
-                          var data = x.data() as Map<String, dynamic>;
-                          Event event = Event.fromMap(data, x.id);
-                          allEvents?.add(event);
+                        if (allEvents == null) {
```
Unfortunately, this I believe makes the view events page less responsive. It does not update when you join an event or subscribe to an event. I tested it on my old code and it does update , but now it does not update anymore.

u22520016 replied:
Okay I'll take a look

u22520016 replied:
I just tested this on another branch it it gives the same issue...The problem is...the user does join the event, but it only shows when the page reloads

MonikaTheiss replied:
On my side when I join an event,it immediatelly adds the event to the upcoming cardsat the end. Same with subscribe. I'm not too sure what branch you ran, but on my side it does immediatelly update when I join/subscribe in the sense tht there is a card added to the upcoming cards/subscribed cards. This is what I meant

MonikaTheiss replied:
I agree that on the event it self the text/icon itself only updates when the page reloads, but the cards are immediatelly supposed to be added when a user joins/subscribes

pieryvdl reviewed (approved):
Hi Franco, I really like the changes you made overall! Looks quite cool and I really enjoy the new homepage. 

Going off what you and Monika discussed: 
1. I think you also missed the appbar, when you click on "See more" in the profile. I'm not quite sure why you did not just use the BackButtonAppBar and adjusted that, instead of replacing the code back to normal appBars, was there a reason why you did? I made that component so that the appBars could be adjusted to avoid the issue of missing appBars. It's okay, just wanted to point it out, maybe you missed it. 

2. I really really like the new homepage and the quick navigation to the search page that you added, I think the reason why it could feel off is because of the white and black? But i really like it
3. I also experienced the bio issue, but not sure if it is on your branch changes, could be an overall issue, so I don't think you have to worry about it too much. 
Overall I really think it is good, I also like the profile page color that you changed.

shan-226 commented:
![gamepage](https://github.com/user-attachments/assets/31756888-950c-48bc-ae58-b7a943e698c4)
![home](https://github.com/user-attachments/assets/763239c0-75de-4640-a04d-97b055f8fd56)
It looks good so far. Just a few small changes. There is an overflow on the home page, i attached a photo above. The font color on the search page is not correct for in light mode, also attached in the above photo. I also have the same issue with the bio. The contrast of the black and white on the home page is also a bit off to me but it does still look good, maybe we can choose softer colors?

u22520016 commented:
> Hi Franco, I really like the changes you made overall! Looks quite cool and I really enjoy the new homepage.
> 
> Going off what you and Monika discussed:
> 
> 1. I think you also missed the appbar, when you click on "See more" in the profile. I'm not quite sure why you did not just use the BackButtonAppBar and adjusted that, instead of replacing the code back to normal appBars, was there a reason why you did? I made that component so that the appBars could be adjusted to avoid the issue of missing appBars. It's okay, just wanted to point it out, maybe you missed it.
> 2. I really really like the new homepage and the quick navigation to the search page that you added, I think the reason why it could feel off is because of the white and black? But i really like it
> 3. I also experienced the bio issue, but not sure if it is on your branch changes, could be an overall issue, so I don't think you have to worry about it too much.
>    Overall I really think it is good, I also like the profile page color that you changed.

Okay yes I forgot about the appbar component Ill change it back

pieryvdl commented:
> Okay yes I forgot about the appbar component Ill change it back

No worries. Just remember to change the app bar component so that it matches what you want those styling to look like. 👍

u22520016 commented:
> Hi @u22520016 , thank you so much for all of this! I am currently running your code on my side here are a few issues I found:
> 
> 1. The appbars are confusing me, the appbar on the message page, updates and invites, and the search page when navigating to it through the homepage are all different.
> 2. Would it perhaps be possible to also , when the clear button on the search component is pressed, that the search stuff is cleared or is that too difficult?
> 3. Little worrisome issue, I cannot update my bio field any more... which seems to be an issue on other branches though
> 4. For the updates and invites page,when you click on it , it first says no updates and then it updates and shows everything.. would it perhaps be possible to rather have a loading screen there?
> 5. It might just be me, but the cards on the home page feel a bit weird to me. Would it perhaps look better if we made the background surface color and then rather add a primary border around the cards? Something on the home page does not sit right with me.
> 
> I really like the changes you made to the view events page and also the changes you made to the game library page👍 let me know what you think of the suggested changes.

The bio issue is again where it does update, but the page has to reload...

MonikaTheiss commented:
> > Hi @u22520016 , thank you so much for all of this! I am currently running your code on my side here are a few issues I found:
> > 
> > 1. The appbars are confusing me, the appbar on the message page, updates and invites, and the search page when navigating to it through the homepage are all different.
> > 2. Would it perhaps be possible to also , when the clear button on the search component is pressed, that the search stuff is cleared or is that too difficult?
> > 3. Little worrisome issue, I cannot update my bio field any more... which seems to be an issue on other branches though
> > 4. For the updates and invites page,when you click on it , it first says no updates and then it updates and shows everything.. would it perhaps be possible to rather have a loading screen there?
> > 5. It might just be me, but the cards on the home page feel a bit weird to me. Would it perhaps look better if we made the background surface color and then rather add a primary border around the cards? Something on the home page does not sit right with me.
> > 
> > I really like the changes you made to the view events page and also the changes you made to the game library page👍 let me know what you think of the suggested changes.
> 
> The bio issue is again where it does update, but the page has to reload...

bio issue you won't have to fix. I know what is happening on there, its a logic error.

u22520016 commented:
I fixed most of the issues you guys told me about:

1. Removed invites updates from home page and moved the event invites to the vent page itself.
2. I fixed (hopefully) all app bars
3. I changed the "reviews" text color so that it is visible
4. I looked at the bio updates and event joining.  The issue is again that the page has to reload for the data to be displayed.  So that is a separate bug that we need to look at in a different branch
